src/test/java/org/apache/maven/plugins/gpg/GpgVersionTest.java

```diff
@@ -19,19 +19,30 @@
  * under the License.
  */
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-import org.apache.maven.plugins.gpg.GpgVersion;
 import org.junit.Test;
 
+/**
+ * Tests for {@link GpgVersion}.
+ */
 public class GpgVersionTest
 {
     @Test
     public void test()
     {
-        assertTrue( GpgVersion.parse( "gpg (GnuPG) 2.2.1" ).isAtLeast( GpgVersion.parse( "gpg (GnuPG) 2.2.1" ) ) );
-        assertTrue( GpgVersion.parse( "gpg (GnuPG) 2.2.1" ).isAtLeast( GpgVersion.parse( "2.1" ) ) );
-        assertTrue( GpgVersion.parse( "gpg (GnuPG/MacGPG2) 2.2.10" ).isAtLeast( GpgVersion.parse( "2.2.10" ) ) );
+        assertEquals( GpgVersion.parse( "gpg (GnuPG) 2.2.1" )
+                .compareTo( GpgVersion.parse( "gpg (GnuPG) 2.2.1" ) ), 0 );
+
+        assertTrue( GpgVersion.parse( "gpg (GnuPG) 2.2.1" )
+                .isAtLeast( GpgVersion.parse( "2.1" ) ) );
+
+        assertEquals( GpgVersion.parse( "gpg (GnuPG/MacGPG2) 2.2.10" )
+                .compareTo( GpgVersion.parse( "2.2.10" ) ), 0 );
+
+        assertEquals( GpgVersion.parse( "gpg (GnuPG) 2.0.26 (Gpg4win 2.2.3)" )
```

Comment:
Why are we testing compareTo here? This should compare two objects directly.

Reply:
Ask Herve, he will tell you that Versions are compared. See the code.

Reply:
@elharo you mean the Object.equals()?

Reply:
I'm the one to blame. `GpgVersion` is a Comparable Object. In the code you're only interest in `before` or `atLeast` some value, you're never interested in equals. This might make the testcode look a bit weird.
Equals compares the raw value, including implementation, and that is not what you want here: the value of the version should be matched.

Reply:
What I meant was these should have been three and now should be four separate test methods with one assertion each. 

The bug here was in parse, so we should be able to test that without using compareTo.

Finally @rfscholte's comment suggests to me the equals method might not be compatible with the comparable method which would be a bug bug if so.

Can the tests be rewritten to more directly address the bug at hand?

Reply:
@rfscholte From the expected business behavior you are right.

>> Equals compares the raw value

The `equals` and `hashcode` is not implemented and it is not the matter of my fix.

I would use `before` and `atLeast` sine this also makes business value for me.
What is missing in the master is the negative test because here we assert that `atLeast` returns `true` but it does not necessarily means that `before` would return `false` in buggy implementation.

Reply:
To me not worth the discussion, it is a matter of style. There was a String that couldn't be parsed correctly, there's a unittest to cover that. So my requirements for the fix are matched.

Reply:
It has been a while since I wrote it, didn't look at the code. Most important: main code needs atLeast or before, you're never interested in the exact value for a version.

Reply:
@rfscholte  Done.

Reply:
If equals isn't implemented, I don't trust the Comparable interface; nor would I trust a unit test based on it.

"The natural ordering for a class C is said to be consistent with equals if and only if e1.compareTo(e2) == 0 has the same boolean value as e1.equals(e2) for every e1 and e2 of class C. Note that null is not an instance of any class, and e.compareTo(null) should throw a NullPointerException even though e.equals(null) returns false.

It is strongly recommended (though not required) that natural orderings be consistent with equals. This is so because sorted sets (and sorted maps) without explicit comparators behave "strangely" when they are used with elements (or keys) whose natural ordering is inconsistent with equals. In particular, such a sorted set (or sorted map) violates the general contract for set (or map), which is defined in terms of the equals method."

Reply:
@elharo 
Therefore @rfscholte mentioned that the requirements matched.
You are technically right regarding the relation compareTo/equals but I would do it in separate commit. Let's review the next commits.